backend/backend.go

```diff
@@ -83,6 +83,24 @@ type Local interface {
 	Context(*Operation) (*terraform.Context, state.State, error)
 }
 
+// Creator can be implemented by backends to automatically create
+// necessary resources during `init`. The backend will be fully configured
```

Comment:
Yeah, I meant it in the general sense of "stuff it needs". My intent here is to reduce friction for new users in the bootstrap process.

I was thinking about tracking the state it somehow, but backends are always a chicken-egg problem in that sense. We can't put it in the backend state since that is ephemeral, and storing remote state on the system that happened to run `init` seems problematic with shared state. Maybe somehow injecting it into the state stored in the backend itself could work, but that also seems likely to create a host of issues to go with it. That road also leads to a backend basically becoming a provider itself.